cost-model/src/transaction_cost.rs

```diff
@@ -1,10 +1,109 @@
-use solana_sdk::pubkey::Pubkey;
+use {crate::block_cost_limits, solana_sdk::pubkey::Pubkey};
+
+/// TransactionCost is used to represent resources required to process
+/// a transaction, denominated in CU (eg. Compute Units).
+/// Resources required to process a regular transaction often include
+/// an array of variables, such as execution cost, loaded bytes, write
+/// lock and read lock etc.
+/// Vote has a simpler and pre-determined format, it's cost structure
+/// can be simpler, calculation quicker.
+#[derive(Debug)]
+pub enum TransactionCost {
+    Vote { writable_accounts: Vec<Pubkey> },
```

Comment:
We know upfront that we will only have 1 or 2 of these writable accounts.

Given the size of the other variant, it's probably better to use a size + array[2] here: `{ num: usize, writable_accounts: [Pubkey; 2] }`.

That way we can avoid an allocation entirely.

Reply:
after #33269 , I am leaning to just give simple_vote a static cost, no need to track vote's writable accounts, their size etc.

Reply:
right, the whole non-vote tx's cu will be apply to the vote account in that scenario, just follow the usual logic. I bet there are some guards to prevent vote embedded in non-vote transaction. Will dig it between tasks ;)

Reply:
So looks like it is always allowed to have _vote ix_ included in normal transaction, where the _vote_ shall be processed as usual. So the scenario you mentioned above is possible. 

As for SimpleVote to use `Vec<Pubkey>` or array of two, imo, using vec is fine for current code since cost-model creates a vec of pubkey for writable accounts, then move to TransactionCost.

(Later when making TransactionCost as part of transaction meta, we can get ride of copying pubkey part, instead just holding write account's index.)